File: DMarket.Client/src/app/shared/models/order.ts
import { Address } from "./applicationUser";

export interface OrderToCreate {
    basketId: string;
    deliveryMethodId: number;
    shipToAddress: Address;
}

export interface Order {
    id: number;
    buyerEmail: string;
    orderDate: string;
    shipToAddress: Address;
    deliveryMethod: string;
    shippingPrice: number;
    orderItems: OrderItem[];
    subtotal: number;
    status: string;
  }

  export interface OrderItem {
    productId: number;
    productName: string;
    pictureUrl: string;
    price: number;
    quantity: number;
  }

File: DMarket.Core/Entities/Order/Order.cs
namespace Core.Entities.Order
{
    public class Order
    {
        public Order()
        {
        }
        public Order(List<OrderItem> orderItems, string buyerEmail, DeliveryAddress shipToAddress, 
            DeliveryMethod deliveryMethod, decimal subtotal)
        {
            BuyerEmail = buyerEmail;
            ShipToAddress = shipToAddress;
            DeliveryMethod = deliveryMethod;
            OrderItems = orderItems;
            Subtotal = subtotal;
        }

        public Guid Id { get; set; } = Guid.NewGuid();
        public string BuyerEmail { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public DeliveryAddress ShipToAddress { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public List<OrderItem> OrderItems { get; set; }
        public decimal Subtotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }
        public decimal Total
        {
            get
            {
                return Subtotal + DeliveryMethod.Price;
            }
        }
    }
}